diff --git a/Sources/Build/BuildDescription/ProductBuildDescription.swift b/Sources/Build/BuildDescription/ProductBuildDescription.swift
index fb78c14baa..7be45e8f37 100644
--- a/Sources/Build/BuildDescription/ProductBuildDescription.swift
+++ b/Sources/Build/BuildDescription/ProductBuildDescription.swift
@@ -310,9 +310,8 @@
                 let macOSSupportedPlatform = self.package.getSupportedPlatform(for: .macOS, usingXCTest: product.isLinkingXCTest)
 
                 if macOSSupportedPlatform.version.major < 12 {
-                    let backDeployedStdlib = try buildParameters.toolchain.macosSwiftStdlib
-                        .parentDirectory
-                        .parentDirectory
+                    let backDeployedStdlib = try AbsolutePath(validating: "@swift-lib@")
+                        .appending("lib")
                         .appending("swift-5.5")
                         .appending("macosx")
                     args += ["-Xlinker", "-rpath", "-Xlinker", backDeployedStdlib.pathString]
diff --git a/Sources/SPMBuildCore/Plugins/DefaultPluginScriptRunner.swift b/Sources/SPMBuildCore/Plugins/DefaultPluginScriptRunner.swift
index 223c60a229..d5a30774a7 100644
--- a/Sources/SPMBuildCore/Plugins/DefaultPluginScriptRunner.swift
+++ b/Sources/SPMBuildCore/Plugins/DefaultPluginScriptRunner.swift
@@ -170,7 +170,7 @@
         }
         else {
             // Add an `-rpath` so the Swift 5.5 fallback libraries can be found.
-            let swiftSupportLibPath = self.toolchain.swiftCompilerPathForManifests.parentDirectory.parentDirectory.appending(components: "lib", "swift-5.5", "macosx")
+            let swiftSupportLibPath = try! AbsolutePath(validating: "@swift-lib@").appending(components: "lib", "swift-5.5", "macosx")
             commandLine += ["-Xlinker", "-rpath", "-Xlinker", swiftSupportLibPath.pathString]
         }
         #endif
diff --git a/Sources/_InternalTestSupport/Toolchain.swift b/Sources/_InternalTestSupport/Toolchain.swift
index 7952ec421e..eafccc0b4a 100644
--- a/Sources/_InternalTestSupport/Toolchain.swift
+++ b/Sources/_InternalTestSupport/Toolchain.swift
@@ -70,7 +70,7 @@
                     try localFileSystem.writeFileContents(inputPath, string: "public func foo() async {}\nTask { await foo() }")
                     let outputPath = tmpPath.appending("foo")
                     let toolchainPath = self.swiftCompilerPath.parentDirectory.parentDirectory
-                    let backDeploymentLibPath = toolchainPath.appending(components: "lib", "swift-5.5", "macosx")
+                    let backDeploymentLibPath = try! AbsolutePath(validating: "@swift-lib@").appending(components: "lib", "swift-5.5", "macosx")
                     try AsyncProcess.checkNonZeroExit(arguments: ["/usr/bin/xcrun", "--toolchain", toolchainPath.pathString, "swiftc", inputPath.pathString, "-Xlinker", "-rpath", "-Xlinker", backDeploymentLibPath.pathString, "-o", outputPath.pathString])
                     try AsyncProcess.checkNonZeroExit(arguments: [outputPath.pathString])
                 }
