diff --git a/swift/stdlib/cmake/modules/StdlibOptions.cmake b/swift/stdlib/cmake/modules/StdlibOptions.cmake
index 24d1b83621..0caeb59a01 100644
--- a/swift/stdlib/cmake/modules/StdlibOptions.cmake
+++ b/swift/stdlib/cmake/modules/StdlibOptions.cmake
@@ -221,9 +221,7 @@
 
    if (SWIFT_ENABLE_DISPATCH)
      set(SWIFT_CONCURRENCY_USES_DISPATCH TRUE)
-     if (NOT CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND NOT EXISTS "${SWIFT_PATH_TO_LIBDISPATCH_SOURCE}")
-       message(SEND_ERROR "Concurrency requires libdispatch on non-Darwin hosts.  Please specify SWIFT_PATH_TO_LIBDISPATCH_SOURCE")
-     endif()
+     find_package(dispatch)
    endif()
 
   if(SWIFT_CONCURRENCY_USES_DISPATCH)
diff --git a/swift/stdlib/public/Concurrency/CMakeLists.txt b/swift/stdlib/public/Concurrency/CMakeLists.txt
index 5f593ecf6f..b728d57a77 100644
--- a/swift/stdlib/public/Concurrency/CMakeLists.txt
+++ b/swift/stdlib/public/Concurrency/CMakeLists.txt
@@ -24,10 +24,7 @@
 set(swift_concurrency_incorporate_object_libraries_so swiftThreading)
 
 if("${SWIFT_CONCURRENCY_GLOBAL_EXECUTOR}" STREQUAL "dispatch")
-  if(NOT CMAKE_SYSTEM_NAME STREQUAL "Darwin")
-    include_directories(AFTER
-                          ${SWIFT_PATH_TO_LIBDISPATCH_SOURCE})
-
+    find_package(dispatch)
     # FIXME: we can't rely on libdispatch having been built for the
     # target at this point in the process.  Currently, we're relying
     # on soft-linking.
